File: Aplicaciones Distribuidas/PARCIAL3/C7_CRUD_VB_GRUPO05/APLICATIVO/CrudLayered_VB/Presentation/Forms/FormEmployee.vb
Imports Domain
Public Class FormEmployee
    Dim employeeModel As New EmployeeModel()
    Private Sub FormEmployee_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        ListEmployee()
        Panel1.Enabled = False
    End Sub
    Private Sub ListEmployee()
        Try
            tableEmployee.DataSource = employeeModel.GetEmployees()
        Catch ex As Exception
            MessageBox.Show(ex.ToString())
        End Try
    End Sub

    Private Sub txtBuscar_TextChanged(sender As Object, e As EventArgs) Handles txtBuscar.TextChanged
        tableEmployee.DataSource = employeeModel.FindById(txtBuscar.Text)
    End Sub

    Private Sub btnAdd_Click(sender As Object, e As EventArgs) Handles btnAdd.Click
        Panel1.Enabled = True
        employeeModel.State = EntityState.Added
    End Sub

    Private Sub btnEdit_Click(sender As Object, e As EventArgs) Handles btnEdit.Click
        If tableEmployee.SelectedRows.Count > 0 Then
            Panel1.Enabled = True
            employeeModel.IdPk = tableEmployee.CurrentRow.Cells(0).Value
            employeeModel.State = EntityState.Modified
            txtIdentificationCard.Text = tableEmployee.CurrentRow.Cells(1).Value
            txtName.Text = tableEmployee.CurrentRow.Cells(2).Value
            txtEmail.Text = tableEmployee.CurrentRow.Cells(3).Value
            dtpBirthday.Text = tableEmployee.CurrentRow.Cells(4).Value
        Else
            MessageBox.Show("Seleccione un registro")
        End If
    End Sub

    Private Sub btnDelete_Click(sender As Object, e As EventArgs) Handles btnDelete.Click
        If tableEmployee.SelectedRows.Count > 0 Then
            Panel1.Enabled = True
            employeeModel.IdPk = tableEmployee.CurrentRow.Cells(0).Value
            employeeModel.State = EntityState.Deleted
            Dim result = employeeModel.SaveChanges()
            MessageBox.Show(result)
            ListEmployee()
        Else
            MessageBox.Show("Seleccione un registro")
        End If
    End Sub

    Private Sub btnSave_Click(sender As Object, e As EventArgs) Handles btnSave.Click
        employeeModel.IdNumber = txtIdentificationCard.Text
        employeeModel.Name = txtName.Text
        employeeModel.Mail = txtEmail.Text
        employeeModel.Birthday = dtpBirthday.Text
        Dim valid = New DataValidation(employeeModel).validate()
        If valid = True Then
            Dim result = employeeModel.SaveChanges()
            MessageBox.Show(result)
            ListEmployee()
            Restart()
        End If
    End Sub
    Private Sub Restart()
        Panel1.Enabled = False
        txtIdentificationCard.Clear()
        txtName.Clear()
        txtEmail.Clear()
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Restart()
    End Sub
End Class

File: Aplicaciones Distribuidas/PARCIAL3/C6_CRUD_C#_GRUPO05/APLICATIVO/LayeredApp/Presentation/Forms/FormEmployee.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Domain.Models;
using Domain.ValueObjects;


namespace Presentation.Forms
{
    public partial class FormEmployee : Form
    {
        private EmployeeModel employee = new EmployeeModel();

        public FormEmployee()
        {
            InitializeComponent();
            panel1.Enabled = false;
        }

        private void label1_Click(object sender, EventArgs e)
        {

        }

        private void textBox1_TextChanged(object sender, EventArgs e)
        {

        }

        private void FormEmployee_Load(object sender, EventArgs e)
        {
            ListEmployees();
        }

        private void ListEmployees()
        {
            try
            {
                dgvEmployee.DataSource = employee.GetAll();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
        }

        private void panel1_Paint(object sender, PaintEventArgs e)
        {

        }

        private void btnSearch_Click(object sender, EventArgs e)
        {
            dgvEmployee.DataSource = employee.FindById(txtSearch.Text);
        }

        private void txtSearch_TextChanged(object sender, EventArgs e)
        {
            dgvEmployee.DataSource = employee.FindById(txtSearch.Text);
        }

        private void btnSave_Click(object sender, EventArgs e)
        {
            employee.IdNumber = txtID.Text;
            employee.Mail = txtEmail.Text;
            employee.Name = txtName.Text;
            employee.Birthday = dtpBirthday.Value;

            bool valid = new Helps.DataValidation(employee).Validate();
            if(valid == true)
            {
                string result = employee.SaveChanges();
                MessageBox.Show(result);
                ListEmployees();
                Restart();
            }
        }

        private void Restart()
        {
            panel1.Enabled = false;
            txtEmail.Clear();  
            txtID.Clear();
            txtName.Clear();
        }


        private void btnNew_Click(object sender, EventArgs e)
        {
            panel1.Enabled = true;
            employee.State = EntityState.Added;
        }

        private void btnEdit_Click(object sender, EventArgs e)
        {
            if(dgvEmployee.SelectedRows.Count > 0)
            {
                panel1.Enabled = true;
                employee.State = EntityState.Modified;
                employee.IdPk = Convert.ToInt32(dgvEmployee.CurrentRow.Cells[0].Value);
                txtID.Text = dgvEmployee.CurrentRow.Cells[1].Value.ToString();
                txtName.Text = dgvEmployee.CurrentRow.Cells[2].Value.ToString();
                txtEmail.Text = dgvEmployee.CurrentRow.Cells[3].Value.ToString();
                dtpBirthday.Value = Convert.ToDateTime(dgvEmployee.CurrentRow.Cells[4].Value);
            }
            else
            {
                MessageBox.Show("Selecet row");
            }
        }

        private void btnRemove_Click(object sender, EventArgs e)
        {
            if (dgvEmployee.SelectedRows.Count > 0)
            {
                employee.State = EntityState.Deleted;
                employee.IdPk = Convert.ToInt32(dgvEmployee.CurrentRow.Cells[0].Value);
                string result = employee.SaveChanges();
                MessageBox.Show(result);
                ListEmployees();
            }
            else
            {
                MessageBox.Show("Selecet row");
            }
        }
    }
}
